### PR TITLE
cabal-install@3.4 3.4.0.0 (new formula)

### PR DESCRIPTION
**Pull request by fishtreesugar**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
https://github.com/Homebrew/homebrew-core/pull/85006#issuecomment-918455294

### CODE REVIEW

File: Formula/cabal-install@3.4.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class CabalInstallAT34 < Formula
+  desc "Command-line interface for Cabal and Hackage"
+  homepage "https://www.haskell.org/cabal/"
+  url "https://hackage.haskell.org/package/cabal-install-3.4.0.0/cabal-install-3.4.0.0.tar.gz"
+  sha256 "1980ef3fb30001ca8cf830c4cae1356f6065f4fea787c7786c7200754ba73e97"
+  license "BSD-3-Clause"
+  keg_only :versioned_formula
```
Comment (carlocab) [inline Formula/cabal-install@3.4.rb:7]:
```suggestion
  license "BSD-3-Clause"

  keg_only :versioned_formula
```

File: Formula/cabal-install@3.4.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  depends_on "ghc"
```
Comment (carlocab) [inline Formula/cabal-install@3.4.rb:9]:
```suggestion
  depends_on "cabal-install" => :build
  depends_on "ghc"
```

File: Formula/cabal-install@3.4.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      sha256 "32d1f7cf1065c37cb0ef99a66adb405f409b9763f14c0926f5424ae408c738ac"
+    end
+  end
+
+  def install
+    resource("bootstrap").stage buildpath
+    cabal = buildpath/"cabal"
+    cd "cabal-install" if build.head?
+    system cabal, "v2-update"
+    system cabal, "v2-install", *std_cabal_v2_args
```
Comment (carlocab) [inline Formula/cabal-install@3.4.rb:34]:
```suggestion
    cd "cabal-install" if build.head?
    system "cabal", "v2-update"
    system "cabal", "v2-install", *std_cabal_v2_args
```

**Comment by fishtreesugar** [comment]
@carlocab thx!

**Comment by SMillerDev** [comment]
Does it meet the requirements in <https://docs.brew.sh/Versions>?

**Comment by fishtreesugar** [comment]
> Versioned formulae submitted should be expected to be used by a large number of people. 

seems no...

I think some niche communities like Haskell, it can't reach the criteria like "used by a large number of people". And the new (3.6) `cabal-install`  broke someone's workflow, bring old (3.4) version back make sth back to normal, that's the purpose of this versioned formula.

**Comment by SMillerDev** [comment]
That one is up to interpretation, I'm more worried about

> Upstream should have a release branch for each formula version, and have an explicit policy of releasing security updates for each version when necessary.

**Comment by fishtreesugar** [comment]
hmm, I think cabal-install seldomly update/backport the old version

**Comment by SMillerDev** [comment]
Then I don't think we can really carry the old version. 

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by fishtreesugar** [comment]
@SMillerDev I see. Thanks for your guidance!

https://github.com/Homebrew/homebrew-core/issues/85133

**Pull request closed without merging**